copy/copy.go

```diff
@@ -646,6 +689,26 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	// If encrypted and decryption keys provided, we should try to decrypt
 	ic.diffIDsAreNeeded = ic.diffIDsAreNeeded || (isEncrypted(src) && ic.c.ociDecryptConfig != nil) || ic.c.ociEncryptConfig != nil
 
+	options.OptimizeDestinationImageAlreadyExists = true // TODO: PR: Remove forced enablement for PR testing
+	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
+	if options.OptimizeDestinationImageAlreadyExists {
+		shouldUpdateSigs := len(sigs) > 0 || options.SignBy != "" // TODO: Consider allowing signatures updates only and skipping the image's layers/manifest copy if possible
+
+		logrus.Debugf("Evaluating image manifest check optimzation: shouldUpdateSigs=%t destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", shouldUpdateSigs, destRequiresOciEncryption, ic.noPendingManifestUpdates())
```

Comment:
```suggestion
		logrus.Debugf("Evaluating image manifest check optimization: shouldUpdateSigs=%t destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", shouldUpdateSigs, destRequiresOciEncryption, ic.noPendingManifestUpdates())
```
at least, or
```suggestion
		logrus.Debugf("Checking if we can avoid the copy: shouldUpdateSigs=%t destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", shouldUpdateSigs, destRequiresOciEncryption, ic.noPendingManifestUpdates())
```
, or even
```suggestion
		logrus.Debugf("Checking if we can avoid the copy: has signatures=%t, OCI encryption=%t, no manifest updates=%t", shouldUpdateSigs, destRequiresOciEncryption, ic.noPendingManifestUpdates())
```
I don’t have a strong preference.

Reply:
sure, I did the latter:
`		logrus.Debugf("Checking if we can avoid the copy: has signatures=%t, OCI encryption=%t, no manifest updates=%t", shouldUpdateSigs, destRequiresOciEncryption, noPendingManifestUpdates)`